### PR TITLE
chore: add shared Node.js configuration

### PR DESCRIPTION
**Pull request by erezrokah**
Adds our shared Node.js project configuration.

Most of the changes are due to `prettier` and `eslint` auto fix.
I'll comment on the ones that aren't

### CODE REVIEW

File: common.js [hunk 1]
```diff
@@ -1,41 +1,35 @@
...
+  /* eslint-disable node/no-unsupported-features/node-builtins */
   if (typeof window !== 'undefined' && window.URL) {
     return new window.URL(url)
+    /* eslint-enable node/no-unsupported-features/node-builtins */
   }
 
+  // eslint-disable-next-line node/global-require
   URLclass = URLclass || require('url')
-  return URLclass.parse(url)
+  return new URLclass.URL(url)
```
Comment (erezrokah) [inline common.js:14]: `url.parse` is deprecated

File: common.js [hunk 2]
```diff
@@ -79,27 +73,24 @@ const exp = {
...
-    } catch (e) {
+    } catch (error) {
       return null
     }
 
     return {
       host: url.host,
       scheme: url.protocol.replace(/:$/, ''),
-      path: url.path,
+      path: url.pathname,
```
Comment (erezrokah) [inline common.js:93]: `new URL()` doesn't have a `path` attribute

File: package.json [hunk 3]
```diff
@@ -18,12 +32,11 @@
     "lodash.isplainobject": "^4.0.6"
   },
   "devDependencies": {
+    "@netlify/eslint-config-node": "^2.2.5",
     "auto-changelog": "^1.13.0",
-    "ava": "^3.6.0",
+    "ava": "^2.4.0",
```
Comment (erezrokah) [inline package.json:37]: Ava v2 to support testing on Node.js 8

**Review by ehmicky — APPROVED** [review]
:broom:

**Pull request merged**